### PR TITLE
Extended version of feature/create news

### PR DESCRIPTION
**Pull request by luchward**
This pull request is based on [Feature/create edit news #646](https://github.com/ita-social-projects/GreenCityClient/pull/646), which has unresolved conflicts.
The base pull request resolves issues [#1584](https://github.com/ita-social-projects/GreenCity/issues/1584), [#660](https://github.com/ita-social-projects/GreenCity/issues/660)
In addition to the base:

- "create news" page has become responsive;
- styles have been fixed to meet mockup requirements;
- the bug, which leads to "Drop your image here or browse" text disapeares when click 'Back to editing' link, has been fixed, [bug#1197](https://github.com/ita-social-projects/GreenCity/issues/1197);
- styles issues have been fixed on "news list gallery" and "news list" pages.

### CODE REVIEW

**Review by skrebetsnataliia — APPROVED** [review]

**Review by Benerous — APPROVED** [review]

**Review by JimHolman1199 — APPROVED** [review]

File: src/app/component/eco-news/directives/drag-and-drop.directive.ts [hunk 1]
```diff
@@ -53,6 +53,7 @@ export class DragAndDropDirective {
     reader.onload = handleFile.bind(this);
 
     function handleFile(event): void {
+      console.log(event);
```
Comment (romanovskyroma) [inline src/app/component/eco-news/directives/drag-and-drop.directive.ts:56]: Do you need this console.log?

**Review by romanovskyroma — APPROVED** [review]

**Review by vitaliipolitylo — APPROVED** [review]

File: src/app/component/eco-news/components/create-edit-news/create-edit-news.component.html [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                <button
+                    class="cancel"
+                    type="button"
+                    (click)="openCancelPopup()">{{ 'create-news.cancel-button' | translate }}</button>
+                <button type="button"
+                        (click)="goToPreview()"
+                >
+                  {{ 'create-news.preview-button' | translate }}
+                </button>
+                <button type="submit" [disabled]="!form.valid ||
```
Comment (mr-pavlenko) [inline src/app/component/eco-news/components/create-edit-news/create-edit-news.component.html:75]: would be good to move this logic to the component from html.

File: src/app/component/eco-news/components/create-edit-news/create-edit-news.component.scss [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+    font-weight: 500;
+    font-size: 48px;
+    line-height: 48px;
+    margin-bottom: 28px;
+  }
+}
+
+.form-description p {
+  max-width: 652px;
+  color: #878787;
```
Comment (mr-pavlenko) [inline src/app/component/eco-news/components/create-edit-news/create-edit-news.component.scss:44]: use variable

File: src/app/component/eco-news/components/create-edit-news/create-edit-news.component.scss [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+}
+
+.date {
+  display: flex;
+  max-width: 500px;
+  margin-top: 32px;
+}
+
+.date p:last-child {
+  margin: 0 0 0 32px;
```
Comment (mr-pavlenko) [inline src/app/component/eco-news/components/create-edit-news/create-edit-news.component.scss:165]: Why do you need 0 0 0 ?

File: src/app/component/eco-news/components/create-edit-news/create-edit-news.component.scss [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+}
+
+.date p span:first-child {
+  font-size: 18px;
+}
+
+.date p span:last-child {
+  color: #878787;
```
Comment (mr-pavlenko) [inline src/app/component/eco-news/components/create-edit-news/create-edit-news.component.scss:173]: use variable

File: src/app/component/eco-news/components/create-edit-news/create-edit-news.component.scss [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+  font-weight: bold;
+}
+
+.submit-buttons button:last-child:disabled {
+  background-color: #839c94;
+  border: 1px solid #839c94;
+}
+
+.tags .filters-color {
+  background-color: #13aa57;
```
Comment (mr-pavlenko) [inline src/app/component/eco-news/components/create-edit-news/create-edit-news.component.scss:211]: the same

File: src/app/component/eco-news/components/create-edit-news/create-edit-news.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,371 @@
...
+import { MAT_DIALOG_DATA, MatDialogRef, MatDialogModule } from '@angular/material/dialog';
+import { MatSnackBarModule } from '@angular/material/snack-bar';
+import { ACTION_CONFIG, ACTION_TOKEN } from './action.constants';
+import { MatSnackBarComponent } from '@global-errors/mat-snack-bar/mat-snack-bar.component';
+import { EcoNewsService } from '@eco-news-service/eco-news.service';
+import { of } from 'rxjs';
+import { EcoNewsModel } from '@eco-news-models/eco-news-model';
+import { FormBuilder } from '@angular/forms';
+import { CreateEditNewsFormBuilder } from './create-edit-news-form-builder';
+import { CreateEcoNewsService } from '@eco-news-service/create-eco-news.service';
```
Comment (mr-pavlenko) [inline src/app/component/eco-news/components/create-edit-news/create-edit-news.component.spec.ts:20]: Don't use services

File: src/app/component/eco-news/components/create-edit-news/create-edit-news.component.spec.ts [hunk 8]
```diff
@@ -0,0 +1,371 @@
...
+import { FormArray, FormControl, FormsModule, ReactiveFormsModule, FormGroup } from '@angular/forms';
+import { DragAndDropComponent } from '@shared/components/drag-and-drop/drag-and-drop.component';
+import { ImageCropperModule } from 'ngx-image-cropper';
+import { RouterTestingModule } from '@angular/router/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { MAT_DIALOG_DATA, MatDialogRef, MatDialogModule } from '@angular/material/dialog';
+import { MatSnackBarModule } from '@angular/material/snack-bar';
+import { ACTION_CONFIG, ACTION_TOKEN } from './action.constants';
+import { MatSnackBarComponent } from '@global-errors/mat-snack-bar/mat-snack-bar.component';
+import { EcoNewsService } from '@eco-news-service/eco-news.service';
```
Comment (mr-pavlenko) [inline src/app/component/eco-news/components/create-edit-news/create-edit-news.component.spec.ts:15]: the same

File: src/app/component/comments/components/comment-pagination/comment-pagination.component.html [hunk 9]
```diff
@@ -4,7 +4,8 @@
                        [maxSize]="calcPaginationSize(p.getLastPage(), p.getCurrent())"
                        (pageChange)="onPageChange($event)"
                        >
-    <div class="custom-pagination">
+    <div class="custom-pagination"
+         *ngIf='p.getTotalItems() > 10'>
```
Comment (mr-pavlenko) [inline src/app/component/comments/components/comment-pagination/comment-pagination.component.html:8]: Magic number 10, You should use a variable for that value.

Comment (skrebetsnataliia) [reply src/app/component/comments/components/comment-pagination/comment-pagination.component.html:8]: It's my mistake. I've fixed it  in this PR https://github.com/ita-social-projects/GreenCityClient/pull/685.

**Pull request merged**